---
 rc |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/rc
+++ b/rc
@@ -41,6 +41,7 @@ startup_progress() {
 startup() {
   # Handle verbosity
   [ "$VERBOSE" = very ] && echo "INIT: Running $@..."
+  logger "INIT: Running $@..."
 
   case "$1" in
 	*.sh)
@@ -56,6 +57,12 @@ startup() {
 		"$@"
 		;;
   esac
+  if [ $? -eq 0 ] ; then
+      logger "INIT: Running $@ PASSED"
+  else
+      logger "INIT: Running $@ FAILED"
+  fi
+
   startup_progress
 }
 
